### DISCUSSION TITLE
How to Deploy a Multitenant SaaS App

### DISCUSSION THREAD

**Discussion by IhsenBouallegue** [argoproj/argo-cd#11465] [Q&A]
I am trying to use ArgoCD to host a SaaS app I developed. I succeeded in writing the manifests and deploying one instance. But as this is a SaaS, I would like this process to be automated.
For example, a user signs up, and the instance is provisioned and exposed under user.app.com.

I have been looking into [ApplicationSets](https://argo-cd.readthedocs.io/en/stable/user-guide/application-set/). It seems to be the way to go, except that I see no generator that supports dynamically adding new "instances/apps". I thought about using the [git config](https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Generators-Git/) generator and appending new values to a config file. But that defies the purpose of automation.

Please I am quite lost, I would appreciate any hint or tip!

**Comment by crenshaw-dev** [comment]
What would you like to be the source of truth for the instances? I assume you'd need an instance name, and maybe some other config?

**Reply by IhsenBouallegue** [reply]
Exactly. And probably the user that created it and other config stuff. But most importantly, the name. So that a specific URL can be generated for it.
I thought at first about saving that config in a database and then triggering the provisioning of a new instance somehow. But I couldn't find a way to dynamically and automatically provision new instances. That's when I started looking into AppSets. But they don't solve my Use Case completely, or maybe I am missing something.
